### DISCUSSION TITLE
Migration from version 0.5/0.6 to latest :  Error level symbol def must be a function returning a component list

### DISCUSSION THREAD

**Discussion by loicdescotte** [replit/kaboom#541] [Q&A] [answered]
Hi,

I am trying to make a little game with kaboom, 
I have forked this tutorial game : https://github.com/kubowania/mario
This game is using kaboom 0.5 

In my version I am trying to use a newer version of kaboom, I have managed to use kaboom 0.6 but when I am trying to use the new version ,  ie if I'm replacing : 
   
 <script src="https://kaboomjs.com/lib/0.6.0/kaboom.js"></script>
by 
    <script src="https://unpkg.com/kaboom/dist/kaboom.js"></script>

I have this error when the game is launching : Error level symbol def must be a function returning a component list

Thanks !

**Comment by slmjkdbtl** [comment, marked as answer]
Instead of 
```js
'x': [
    sprite(),
    pos(),
    // other components..
]
```
you should pass a function that returns the list of components
```js
'x': () => [
    sprite(),
    pos(),
    // other components..
]
```
which is a shorthand for
```js
'x': () => {
    return [
        sprite(),
        pos(),
        // other components..
    ]
}
```

**Reply by loicdescotte** [reply]
thanks
